gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-sync/src/main/java/io/gravitee/gateway/services/sync/process/common/deployer/SubscriptionDeployer.java

```diff
@@ -180,24 +181,41 @@ private Completable sendFailureCommand(Subscription subscription, Throwable thro
                 command.setUpdatedAt(Date.from(now));
                 command.setEnvironmentId(subscription.getEnvironmentId());
 
-                convertSubscriptionCommand(subscription, command, throwable.getMessage());
+                convertSubscriptionCommand(subscription, command, throwable);
 
                 saveCommand(subscription, command);
             })
             .subscribeOn(Schedulers.io());
     }
 
-    private void convertSubscriptionCommand(Subscription subscription, Command command, String errorMessage) {
+    private void convertSubscriptionCommand(Subscription subscription, Command command, Throwable throwable) {
+        var failureCause = extractFailureCause(throwable);
         try {
-            command.setContent(objectMapper.writeValueAsString(new SubscriptionFailureCommand(subscription.getId(), errorMessage)));
+            command.setContent(objectMapper.writeValueAsString(new SubscriptionFailureCommand(subscription.getId(), failureCause)));
         } catch (JsonProcessingException e) {
             log.error("Failed to convert subscription command [{}] to string", subscription.getId(), e);
             JsonObject json = new JsonObject();
-            json.put("subscriptionId", subscription.getId()).put("failureCause", errorMessage);
+            json.put("subscriptionId", subscription.getId()).put("failureCause", failureCause);
             command.setContent(json.encode());
         }
     }
 
+    private String extractFailureCause(Throwable throwable) {
+        if (throwable instanceof InterruptionFailureException interruptionFailureException) {
+            var executionFailure = interruptionFailureException.getExecutionFailure();
+            var cause = (Throwable) executionFailure.parameters().get("exception");
+            var detailedMessage = (String) executionFailure.parameters().get("detailedMessage");
+
+            var message = executionFailure.message();
+            if (detailedMessage != null) {
+                return message + "\n" + detailedMessage;
+            }
+            return message + '\n' + cause.getMessage();
```

Comment:
I know that is verbose, but I have PTSD of cast... Maybe we can do:
```suggestion
            var causeMessage = executionFailure.parameters().get("exception") instanceof Throwable th
                ? th.getMessage()
                : null;

            var message = executionFailure.message();
            return executionFailure.parameters().get("detailedMessage") instanceof String detailedMessage
                ? message + '\n' + detailedMessage
                : message + '\n' + causeMessage;
```